workspace/resource_notebook.go

```diff
@@ -431,7 +442,13 @@ func ResourceNotebook() *schema.Resource {
 			})
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			return NewNotebooksAPI(ctx, c).Delete(d.Id(), true)
+			objType := d.Get("object_type")
+			return NewNotebooksAPI(ctx, c).Delete(d.Id(), !(objType == Notebook || objType == File))
```

Comment:
So, if objType is Directory, Library, or Repo, we do want to recursively delete?

Reply:
it's only Directory that could be here.